### PR TITLE
"Deconstruct variable declaration" should not introduce errors

### LINKED ISSUE #25260: "Deconstruct variable declaration" introduces error.
Start with code:

```c#
        (int x, int y) t = default;
        Console.WriteLine(t.x + t.y);
```

Use 'Deconstruct variable declaration' on 't'.  It produces the following code:

```c#
        (int x, int y) = default;
        Console.WriteLine(x + y);
```

This is not allowed because 'default' is not deconstructable to a tuple.

### PR DESCRIPTION
**Pull request by Neme12**
fixes #25260

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/UseDeconstruction/UseDeconstructionTests.cs [hunk 1]
```diff
@@ -465,6 +465,201 @@ void M()
...
+        [WorkItem(25260, "https://github.com/dotnet/roslyn/issues/25260")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseDeconstruction)]
+        public async Task TestNotWithDefaultLiteralInitializer()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        (string name, int age) [|person|] = default;
```
Comment (jcouv) [inline src/EditorFeatures/CSharpTest/UseDeconstruction/UseDeconstructionTests.cs]: Filed https://github.com/dotnet/roslyn/issues/25559

**Comment by jcouv** [comment]
@dotnet/roslyn-ide Please review and merge this community PR. Thanks

**Comment by jcouv** [comment]
@Pilchie Do we need some kind of ask-mode approval for master? If yes, could you approve?

**Comment by jinujoseph** [comment]
@jcouv no master doesn't need ask mode approval.
I was waiting for test to complete

**Comment by Neme12** [comment]
I don't know what's going on :/ it says `WorkItemAttribute` can't be found, but it's building fine for me locally.

**Comment by jinujoseph** [comment]
confirmed with @jasonmalinowski , known issue ..merging this

**Comment by Neme12** [comment]
Wait, it seems like a different issue, `using Roslyn.Test.Utilities` must have been removed from the file, even though it's still shown in the diff.

**Comment by Neme12** [comment]
I'm afraid this is going to break the build :-/

**Comment by Neme12** [comment]
@jcouv @jinujoseph @jasonmalinowski I'm afraid this is going to break the build on master :-/ This was a real build issue, the using was removed on master

I created a PR to fix it #25689
I'm really sorry :-/

**Comment by sharwell** [comment]
@Neme12 No problem, thanks for the fix 😄

**Comment by jasonmalinowski** [comment]
Don't worry, we award our build break squirrel to whoever hit the merge button, so @jinujoseph now gets to keep it company.

**Comment by jcouv** [comment]
I've checked out the latest bits from master after this PR was merged and I'm indeed seeing the build break.

**Comment by jinujoseph** [comment]
@Neme12  don't worry ...it was my mistake , i deserve the squirrel , thanks for the fix

**Comment by Neme12** [comment]
@jinujoseph no it's my fault I commented too quickly without thinking about it, I'm sorry

**Pull request merged**